tools/sphere.c

```diff
@@ -80,3 +78,31 @@ int do_touch( string str ) {
 int do_gaze( string str ) {
 
 }
+
+/**
```

Comment:
This is great, exactly what you should be communicating in your API docs. Just a few notes:

* Remove the line of whitespace between the comment and the function declaration.
* You don't need to put the {Type} stuff. That's a carryover over from JavaScript docs which don't use explicit types in their member declarations. I.e. our function declaration tells us it's a string, so we don't need it in our doc.
* Don't bother with the colons before the description text. Parameter names can't contain whitespace, so we know the description will always start after the first word.
* Your return description says we return the shadow object, which isn't true. Right now this function is returning the program name of the shadow object, a string.
* You should mention that the function returns 0 if no matching shadow is found.